src/orquestra/opt/optimizers/pso/continuous_pso_optimizer.py

```diff
@@ -0,0 +1,333 @@
+from typing import Callable, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import scipy
+from scipy.optimize import OptimizeResult
+
+from orquestra.opt.api import (
+    CostFunction,
+    Optimizer,
+    construct_history_info,
+    optimization_result,
+)
+from orquestra.opt.history.recorder import RecorderFactory, recorder as _recorder
+from orquestra.opt.optimizers.pso.topologies import StarTopology, SwarmTopology
+
+Bounds = Union[
+    scipy.optimize.Bounds, Sequence[Tuple[float, float]], Tuple[float, float]
+]
+
+
+def _get_bounds_like_array(
+    bounds: Bounds,
+) -> Union[np.ndarray, Tuple[np.ndarray, np.ndarray]]:
+    """
+    Casts a Bounds object to an object with two entries: the first being the lower
+    bounds and the second being the upper bounds.
+
+    Parameters
+    ----------
+    bounds : Bounds
+        A Bounds object, which can be a scipy.optimize.Bounds object, a sequence of
+        tuples, one tuple being the bonds per parameter, or a tuple of two floats,
+        which are the lower and upper bounds for all parameters.
+
+    Returns
+    -------
+    Union[np.ndarray, Tuple[np.ndarray, np.ndarray]]
+        Lower and upper bounds.
+    """
+    if isinstance(bounds, scipy.optimize.Bounds):
```

Comment:
Also, it seems that bounds are used only in init, so perhaps we don't need this function at all.
What about having a function `extract_scale_and_shift(bounds: Bounds)` ?

Reply:
I have improved the output to be the union of (float, float) or (np.ndarray, np.ndarray). Basically, one uses those bounds to perform arithmetic operations, so broadcasting takes care of the case where the output is (float, float).
Since the function is used for velocity bounds as well (and we explicitly use those velocity bounds for some operations elsewhere), we should keep this function and not define `extract_scale_and_shift` as a complete block for treating the `Bounds` object.